model/implicit_model.py

```diff
@@ -63,3 +63,6 @@ def get_variables(self):
     def get_objective_value(self):
         """ Returns the object value of the found solution """
         return self.model.getObjective().getValue()
+
+    def save_solution(self, filename):
+        self.model.write(f"solutions/{filename}-ESP.sol")
```

Comment:
Care to change this to `self.model.write(f"solutions/{filename}-IMPLICIT.sol")`?